vault-docker/Dockerfile

```diff
@@ -1,6 +1,40 @@
-FROM hashicorp/vault:1.6.2
+FROM --platform=$BUILDPLATFORM hashicorp/vault:1.8.0
 
-RUN wget https://github.com/heronimus/ct-plugin-string2files/releases/download/v0.1.1/string2files-v0.1.1-linux-amd64-musl-debug
-RUN chmod +x string2files-v0.1.1-linux-amd64-musl-debug
-RUN mv string2files-v0.1.1-linux-amd64-musl-debug /usr/local/bin/string2files
+ARG TARGETPLATFORM
+ARG TARGETOS
+ARG TARGETARCH
+ARG DOCKER_USERNAME
+ARG DOCKER_PASSWORD
 
+
+# Install necessary dependencies
+RUN apk update && apk add --no-cache \
+    wget \
+    git \
+    docker \
+    build-base
+
+# Use the build-time variables to authenticate with the registry
+RUN echo "$DOCKER_PASSWORD" | docker login --username "$DOCKER_USERNAME" --password-stdin
+
+# Clone the V programming language repository
+RUN git clone https://github.com/vlang/v \
+    && cd v \
+    && make
+
+# Symlink the V binary
+RUN ln -s /v/v /usr/local/bin/v
+
+# Clone the ct-plugin-string2files repository
+RUN git clone https://github.com/heronimus/ct-plugin-string2files.git \
```

Comment:
Can we clone from a fork in our organization?

Reply:
Yes sure we can do that 👍

Reply:
I kept this, because if there are updates on the upstream we will get those as well and we wont need to update it